diff --git a/base/BUILD.gn b/base/BUILD.gn
index 82cb596dcbb369cf67addf856dfe8c0789449e78..e157aa9e2149f97b8df2db33dc7a5710559e9d8d 100644
--- a/base/BUILD.gn
+++ b/base/BUILD.gn
@@ -4461,6 +4461,7 @@ if (is_android) {
       "android/java/src/org/chromium/base/task/ThreadPoolTaskExecutor.java",
       "android/java/src/org/chromium/base/task/UiThreadTaskExecutor.java",
     ]
+    sources += brave_java_base_sources
 
     if (use_clang_profiling) {
       sources += [
@@ -4441,6 +4442,7 @@ if (is_android) {
       "android/proguard/chromium_apk.flags",
       "android/proguard/chromium_code.flags",
       "android/proguard/remove_logging.flags",
+      "//brave/third_party/sentry/sentry-proguard.pro",
     ]
   }
 
@@ -4500,6 +4501,7 @@ if (is_android) {
       "android/java/src/org/chromium/base/shared_preferences/SharedPreferencesManager.java",
       "android/java/src/org/chromium/base/shared_preferences/StrictPreferenceKeyChecker.java",
     ]
+    sources += brave_base_shared_preferences_java_sources
   }
 
   android_aidl("process_launcher_aidl") {
