diff --git a/base/BUILD.gn b/base/BUILD.gn
index 24f2397278ccb400a47e065a676be5ae783fa2e6..1727a90cfe311bab5115f4579783f6bb5593c3f9 100644
--- a/base/BUILD.gn
+++ b/base/BUILD.gn
@@ -4480,6 +4480,7 @@ if (is_android) {
       "android/java/src/org/chromium/base/task/ThreadPoolTaskExecutor.java",
       "android/java/src/org/chromium/base/task/UiThreadTaskExecutor.java",
     ]
+    sources += brave_java_base_sources
 
     if (use_clang_profiling) {
       sources += [
@@ -4495,6 +4496,7 @@ if (is_android) {
       "android/proguard/chromium_apk.flags",
       "android/proguard/chromium_code.flags",
       "android/proguard/remove_logging.flags",
+      "//brave/third_party/sentry/sentry-proguard.pro",
     ]
   }
 
@@ -4515,6 +4517,7 @@ if (is_android) {
       "android/java/src/org/chromium/base/shared_preferences/SharedPreferencesManager.java",
       "android/java/src/org/chromium/base/shared_preferences/StrictPreferenceKeyChecker.java",
     ]
+    sources += brave_base_shared_preferences_java_sources
   }
 
   android_aidl("process_launcher_aidl") {
