diff --git a/chrome/android/java/src/org/chromium/chrome/browser/contextmenu/ChromeContextMenuPopulator.java b/chrome/android/java/src/org/chromium/chrome/browser/contextmenu/ChromeContextMenuPopulator.java
<<<<<<< HEAD
index 14bd7290e5c3a1b8a9b805ebbdf3b82f9719a1fb..9c1fdc8516a8d85823e880906b14fd633541b1b5 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/contextmenu/ChromeContextMenuPopulator.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/contextmenu/ChromeContextMenuPopulator.java
@@ -133,7 +133,7 @@ public class ChromeContextMenuPopulator implements ContextMenuPopulator {
                 Action.COPY_IMAGE, Action.SHOP_IMAGE_WITH_GOOGLE_LENS, Action.READ_LATER,
                 Action.SHOP_WITH_GOOGLE_LENS_CHIP, Action.TRANSLATE_WITH_GOOGLE_LENS_CHIP,
                 Action.SHARE_HIGHLIGHT, Action.REMOVE_HIGHLIGHT, Action.LEARN_MORE,
-                Action.OPEN_IN_NEW_TAB_IN_GROUP})
+                Action.OPEN_IN_NEW_TAB_IN_GROUP, Action.COPY_CLEAN_LINK})
         @Retention(RetentionPolicy.SOURCE)
         public @interface Action {
             int OPEN_IN_NEW_TAB = 0;
@@ -177,7 +177,7 @@ public class ChromeContextMenuPopulator implements ContextMenuPopulator {
=======
index e87f7da1bceadace32c0d9f17dd62e345e036197..292439d13d8841312058204b4a7dffaf688725f8 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/contextmenu/ChromeContextMenuPopulator.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/contextmenu/ChromeContextMenuPopulator.java
@@ -153,7 +153,7 @@ public class ChromeContextMenuPopulator implements ContextMenuPopulator {
             Action.SHARE_HIGHLIGHT,
             Action.REMOVE_HIGHLIGHT,
             Action.LEARN_MORE,
-            Action.OPEN_IN_NEW_TAB_IN_GROUP
+            Action.OPEN_IN_NEW_TAB_IN_GROUP, Action.COPY_CLEAN_LINK
         })
         @Retention(RetentionPolicy.SOURCE)
         public @interface Action {
@@ -198,7 +198,7 @@ public class ChromeContextMenuPopulator implements ContextMenuPopulator {
>>>>>>> parent of d8c6a33798f (Revert "Upgrade Chromium from 120 to 121")
             int LEARN_MORE = 38;
             int OPEN_IN_NEW_TAB_IN_GROUP = 39;
             int OPEN_IN_NEW_WINDOW = 40;
-            int NUM_ENTRIES = 41;
+            int COPY_CLEAN_LINK = 41;int NUM_ENTRIES = 42;
         }
     }
 
<<<<<<< HEAD
@@ -237,7 +237,7 @@ public class ChromeContextMenuPopulator implements ContextMenuPopulator {
 
         if (mParams.isAnchor()) {
             ModelList linkGroup = new ModelList();
-            if (FirstRunStatus.getFirstRunFlowComplete() && !isEmptyUrl(mParams.getUrl())
+            if (!isEmptyUrl(mParams.getUrl())
                     && UrlUtilities.isAcceptedScheme(mParams.getUrl())) {
                 if (mMode == ContextMenuMode.NORMAL) {
                     linkGroup.add(createListItem(Item.OPEN_IN_NEW_TAB_IN_GROUP));
@@ -307,7 +307,7 @@ public class ChromeContextMenuPopulator implements ContextMenuPopulator {
             }
         }
 
-        if (mParams.isImage() && FirstRunStatus.getFirstRunFlowComplete()) {
+        if (mParams.isImage()) {
             ModelList imageGroup = new ModelList();
             boolean isSrcDownloadableScheme =
                     UrlUtilities.isDownloadableScheme(mParams.getSrcUrl());
@@ -371,7 +371,7 @@ public class ChromeContextMenuPopulator implements ContextMenuPopulator {
             groupedItems.add(new Pair<>(R.string.contextmenu_image_title, imageGroup));
         }
 
-        if (mParams.isVideo() && FirstRunStatus.getFirstRunFlowComplete() && mParams.canSaveMedia()
+        if (mParams.isVideo() && mParams.canSaveMedia()
                 && UrlUtilities.isDownloadableScheme(mParams.getSrcUrl())) {
             ModelList videoGroup = new ModelList();
             videoGroup.add(createListItem(Item.SAVE_VIDEO));
@@ -390,7 +390,7 @@ public class ChromeContextMenuPopulator implements ContextMenuPopulator {
             groupedItems.add(new Pair<>(null, sharedHighlightingGroup));
         }
 
-        if (mMode != ContextMenuMode.NORMAL && FirstRunStatus.getFirstRunFlowComplete()) {
+        if (mMode != ContextMenuMode.NORMAL) {
             ModelList items = groupedItems.isEmpty()
                     ? new ModelList()
                     : groupedItems
=======
@@ -287,6 +287,7 @@ public class ChromeContextMenuPopulator implements ContextMenuPopulator {
             if (!MailTo.isMailTo(mParams.getLinkUrl().getSpec())
                     && !UrlUtilities.isTelScheme(mParams.getLinkUrl())) {
                 linkGroup.add(createListItem(Item.COPY_LINK_ADDRESS));
+                linkGroup.add(createListItem(Item.COPY_CLEAN_LINK));
                 if (!mParams.getLinkText().trim().isEmpty() && !mParams.isImage()) {
                     linkGroup.add(createListItem(Item.COPY_LINK_TEXT));
                 }
>>>>>>> parent of d8c6a33798f (Revert "Upgrade Chromium from 120 to 121")
