diff --git a/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java b/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
index 410e289c3ceefb1423d5966bd705492301051483..09ceb974f472fa48234330c83f55675dc8833ef8 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
@@ -815,6 +815,13 @@ public class ToolbarManager implements UrlFocusChangeListener, ThemeColorObserve
                 updateTabLoadingState(true);
                 onBackPressStateChanged();
 
+                // NewTabPage ntp = getNewTabPageForCurrentTab();
+                // if (ntp != null) {
+                //     mLocationBar.getContainerView().setVisibility(View.GONE);
+                // } else {
+                //     mLocationBar.getContainerView().setVisibility(View.VISIBLE);
+                // }
+
                 // A URL update is a decent enough indicator that the toolbar widget is in
                 // a stable state to capture its bitmap for use in fullscreen.
                 mControlContainer.setReadyForBitmapCapture(true);
@@ -1796,6 +1803,18 @@ public class ToolbarManager implements UrlFocusChangeListener, ThemeColorObserve
     public void onUrlFocusChange(boolean hasFocus) {
         mToolbar.onUrlFocusChange(hasFocus);
 
+        // if (hasFocus) {
+        //     mLocationBar.getContainerView().setVisibility(View.VISIBLE);
+        // } else {
+        //     NewTabPage ntp = getNewTabPageForCurrentTab();
+        //     if (ntp != null) {
+        //         mLocationBar.getContainerView().setVisibility(View.GONE);
+        //     } else {
+        //         mLocationBar.getContainerView().setVisibility(View.VISIBLE);
+        //     }
+        // }
+        
+
         if (mIsStartSurfaceRefactorEnabled && mLayoutStateProvider != null
                 && mLayoutStateProvider.isLayoutVisible(LayoutType.START_SURFACE)) {
             mToolbar.updateStartSurfaceToolbarState(null, !hasFocus, LayoutType.START_SURFACE);
