diff --git a/chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunFlowSequencer.java b/chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunFlowSequencer.java
index 1d05160303033f056ba9c5bfa08acc781a4aebbd..bc6067f66086a223cd874a28693df7cc8a15f35d 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunFlowSequencer.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunFlowSequencer.java
@@ -226,11 +226,12 @@ public abstract class FirstRunFlowSequencer {
      */
     public static boolean checkIfFirstRunIsNecessary(
             boolean preferLightweightFre, Intent fromIntent) {
-        boolean isCct =
-                fromIntent.getBooleanExtra(
-                                FirstRunActivityBase.EXTRA_CHROME_LAUNCH_INTENT_IS_CCT, false)
-                        || LaunchIntentDispatcher.isCustomTabIntent(fromIntent);
-        return checkIfFirstRunIsNecessary(preferLightweightFre, isCct);
+        return false;
+        // boolean isCct =
+        //         fromIntent.getBooleanExtra(
+        //                         FirstRunActivityBase.EXTRA_CHROME_LAUNCH_INTENT_IS_CCT, false)
+        //                 || LaunchIntentDispatcher.isCustomTabIntent(fromIntent);
+        // return checkIfFirstRunIsNecessary(preferLightweightFre, isCct);
     }
 
     /**
