diff --git a/chrome/app/chrome_main_delegate.cc b/chrome/app/chrome_main_delegate.cc
index 4c150b5cb007dbd8cb3ea5078fa3043fcccc6370..84284160402129276408905a341ede634fb7d31b 100644
--- a/chrome/app/chrome_main_delegate.cc
+++ b/chrome/app/chrome_main_delegate.cc
@@ -100,6 +100,7 @@
 #include "ui/base/resource/resource_bundle.h"
 #include "ui/base/resource/scoped_startup_resource_bundle.h"
 #include "ui/base/ui_base_switches.h"
+#include "brave/third_party/sentry_native/include/sentry.h"
 
 #if BUILDFLAG(IS_WIN)
 #include <malloc.h>
@@ -1604,6 +1605,24 @@ void ChromeMainDelegate::PreSandboxStartup() {
     if (process_type.empty()) {
       base::android::InitJavaExceptionReporter();
       UninstallPureJavaExceptionHandler();
+
+      sentry_options_t* options = sentry_options_new();
+      sentry_options_set_dsn(
+          options,
+          "https://591d5ee7a98b1fdeca311066ff238573@o4509807793733632.ingest.de.sentry.io/4509807795372112");
+      sentry_options_set_release(options, "chromium@");
+      sentry_options_set_environment(options, "production");
+
+      // Disable Sentry's own crash handler to avoid conflicts with Crashpad
+      sentry_options_set_system_crash_reporter_enabled(options, 1);
+
+      base::FilePath path;
+      if (base::PathService::Get(base::DIR_ANDROID_APP_DATA, &path)) {
+          // The path needs to be a C-style string.
+          sentry_options_set_database_path(options, path.value().c_str());
+      }
+
+      sentry_init(options);
     } else {
       base::android::InitJavaExceptionReporterForChildProcess();
     }
