diff --git a/chrome/app/BUILD.gn b/chrome/app/BUILD.gn
index 7b5a8e56879b7aab545da6ca847d809020e11e96..c8da34ac0aec7fbc240a3be7816bc9ca9121a17f 100644
--- a/chrome/app/BUILD.gn
+++ b/chrome/app/BUILD.gn
@@ -117,6 +117,7 @@ grit("generated_resources") {
   if (is_android) {
     outputs += android_generated_java_resources
   }
+  deps = [ "//brave/app:brave_generated_resources_grit" ]
 }
 
 if (is_android) {
@@ -180,6 +181,7 @@ static_library("test_support") {
     "//services/tracing/public/cpp",
     "//ui/base",
     "//v8:v8_headers",
+    "//brave/third_party/sentry_native:sentry_native",
   ]
   public_deps = []
 
