diff --git a/chrome/browser/ui/android/theme/java/src/org/chromium/chrome/browser/theme/ThemeUtils.java b/chrome/browser/ui/android/theme/java/src/org/chromium/chrome/browser/theme/ThemeUtils.java
index 69072319afa8abb16184ab6f7fdbf8065fee5b9d..01a819ec36b3ffd4af8c63cf81717c8b1154d58e 100644
--- a/chrome/browser/ui/android/theme/java/src/org/chromium/chrome/browser/theme/ThemeUtils.java
+++ b/chrome/browser/ui/android/theme/java/src/org/chromium/chrome/browser/theme/ThemeUtils.java
@@ -113,7 +113,7 @@ public class ThemeUtils {
      * @return The {@link ColorStateList} for the icon tint of themed toolbar.
      */
     public static ColorStateList getThemedToolbarIconTint(Context context, boolean useLight) {
-        return AppCompatResources.getColorStateList(context, getThemedToolbarIconTintRes(useLight));
+        return AppCompatResources.getColorStateList(context, getThemedToolbarIconTintRes(false));
     }
 
     /**
@@ -159,6 +159,7 @@ public class ThemeUtils {
         } else if (brandedColorScheme == BrandedColorScheme.DARK_BRANDED_THEME) {
             colorId = R.color.default_icon_color_white_tint_list;
         }
+        colorId = R.color.default_icon_color_white_tint_list;
         return colorId;
     }
 
