diff --git a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarPhone.java b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarPhone.java
index d54918306cd23c35c6235a5f95d0c35a39d3af83..256a112500caceb6c4a644472209cc9ce8d0a8a6 100644
--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarPhone.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/top/ToolbarPhone.java
@@ -392,7 +392,7 @@ public class ToolbarPhone extends ToolbarLayout
         getToolbarDataProvider().addToolbarDataProviderObserver(this);
 
         mHomeButtonDisplay.updateState(
-                mVisualState, mIsHomeButtonEnabled, mIsHomepageNonNtp, urlHasFocus());
+                mVisualState, true, mIsHomepageNonNtp, urlHasFocus());
     }
 
     @Override
@@ -460,6 +460,7 @@ public class ToolbarPhone extends ToolbarLayout
         if (mOptionalButtonCoordinator != null) {
             mOptionalButtonCoordinator.setBackgroundColorFilter(color);
         }
+        BraveToolbarLayout.class.cast(this).updateModernLocationBarColorImpl(color);
     }
 
     private void updateModernLocationBarCorners() {
@@ -740,6 +741,8 @@ public class ToolbarPhone extends ToolbarLayout
             padding = mHomeButtonDisplay.getMeasuredWidth();
         }
 
+        mHomeButtonDisplay.setVisibility(INVISIBLE);
+
         return padding;
     }
 
@@ -998,7 +1001,7 @@ public class ToolbarPhone extends ToolbarLayout
         int toolbarButtonVisibility = getToolbarButtonVisibility();
         mToolbarButtonsContainer.setVisibility(toolbarButtonVisibility);
         if (mHomeButtonDisplay.getVisibility() != GONE) {
-            mHomeButtonDisplay.setVisibility(toolbarButtonVisibility);
+            mHomeButtonDisplay.setVisibility(INVISIBLE);
         }
 
         updateLocationBarLayoutForExpansionAnimation();
@@ -1016,6 +1019,7 @@ public class ToolbarPhone extends ToolbarLayout
      * Tab Page.
      */
     private void updateLocationBarLayoutForExpansionAnimation() {
+        if (!BraveToolbarLayout.class.cast(this).isLocationBarValid(mLocationBar)) return;
         TraceEvent.begin("ToolbarPhone.updateLocationBarLayoutForExpansionAnimation");
         if (isInTabSwitcherMode()) return;
 
@@ -1394,6 +1398,7 @@ public class ToolbarPhone extends ToolbarLayout
             canvas.restore();
         }
 
+        BraveToolbarLayout.class.cast(this).drawAnimationOverlay(mToolbarButtonsContainer, canvas);
         // Draw the tab stack button and associated text if necessary.
         if (getTabSwitcherButtonCoordinator() != null && mUrlExpansionFraction != 1f) {
             // Draw the tab stack button image.
@@ -1704,6 +1709,7 @@ public class ToolbarPhone extends ToolbarLayout
     @Override
     public void onHomeButtonIsEnabledUpdate(boolean homeButtonEnabled) {
         mIsHomeButtonEnabled = homeButtonEnabled;
+        mIsHomeButtonEnabled = false;
         updateButtonVisibility();
     }
 
@@ -1723,7 +1729,7 @@ public class ToolbarPhone extends ToolbarLayout
     public void updateButtonVisibility() {
         if (mHomeButtonDisplay != null) {
             mHomeButtonDisplay.updateState(
-                    mVisualState, mIsHomeButtonEnabled, mIsHomepageNonNtp, urlHasFocus());
+                    mVisualState, true, mIsHomepageNonNtp, urlHasFocus());
         }
     }
 
@@ -2043,6 +2049,7 @@ public class ToolbarPhone extends ToolbarLayout
         } else {
             populateUrlClearExpansionAnimatorSet(animators);
         }
+        if (!BraveToolbarLayout.class.cast(this).isLocationBarValid(mLocationBar)) return;
         mUrlFocusLayoutAnimator = new AnimatorSet();
         mUrlFocusLayoutAnimator.playTogether(animators);
 
@@ -2461,7 +2468,7 @@ public class ToolbarPhone extends ToolbarLayout
         mVisualState = newVisualState;
 
         mHomeButtonDisplay.updateState(
-                mVisualState, mIsHomeButtonEnabled, mIsHomepageNonNtp, urlHasFocus());
+                mVisualState, true, mIsHomepageNonNtp, urlHasFocus());
 
         // Refresh the toolbar texture.
         if ((mVisualState == VisualState.BRAND_COLOR || visualStateChanged)
