diff --git a/chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunFlowSequencer.java b/chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunFlowSequencer.java
index 7c5e4e417483d71a18c76408047b5b2dfbb9499a..7cc46d56ca9434b7a65aad11139ec24b9e4d303a 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunFlowSequencer.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunFlowSequencer.java
@@ -217,10 +217,11 @@ public abstract class FirstRunFlowSequencer  {
      */
     public static boolean checkIfFirstRunIsNecessary(
             boolean preferLightweightFre, Intent fromIntent) {
-        boolean isCct = fromIntent.getBooleanExtra(
-                                FirstRunActivityBase.EXTRA_CHROME_LAUNCH_INTENT_IS_CCT, false)
-                || LaunchIntentDispatcher.isCustomTabIntent(fromIntent);
-        return checkIfFirstRunIsNecessary(preferLightweightFre, isCct);
+        return false;
+        // boolean isCct = fromIntent.getBooleanExtra(
+        //                         FirstRunActivityBase.EXTRA_CHROME_LAUNCH_INTENT_IS_CCT, false)
+        //         || LaunchIntentDispatcher.isCustomTabIntent(fromIntent);
+        // return checkIfFirstRunIsNecessary(preferLightweightFre, isCct);
     }
 
     /**
