diff --git a/chrome/app/BUILD.gn b/chrome/app/BUILD.gn
index 807530111d7c4af45ea4f0efcf3a1cea9f2e2256..6fc8a51cf2ffa7bf86db8b1ac33849f6d7df9bea 100644
--- a/chrome/app/BUILD.gn
+++ b/chrome/app/BUILD.gn
@@ -117,6 +117,7 @@ grit("generated_resources") {
   if (is_android) {
     outputs += android_generated_java_resources
   }
+  deps = [ "//brave/app:brave_generated_resources_grit" ]
 }
 
 if (is_android) {
@@ -180,6 +181,7 @@ static_library("test_support") {
     "//services/tracing/public/cpp",
     "//ui/base",
     "//v8:v8_headers",
+    "//brave/third_party/sentry_native:sentry_native",
   ]
   public_deps = []
 
