diff --git a/components/search_engines/default_search_manager.cc b/components/search_engines/default_search_manager.cc
index 1771bd789b04ceb9afb78290f4f25bc19be55c83..0b8db8e3b4ce9e737b51fcc0c5bf634a2d8393b1 100644
--- a/components/search_engines/default_search_manager.cc
+++ b/components/search_engines/default_search_manager.cc
@@ -355,6 +355,7 @@ void DefaultSearchManager::LoadDefaultSearchEngineFromPrefs() {
   // Check if default search preference is overriden by extension.
   if (pref->IsExtensionControlled()) {
     extension_default_search_ = std::move(turl_data);
+  // LOADDEFAULTSEARCHENGINEFROMPREFS_BRAVE
   } else {
     prefs_default_search_ = std::move(turl_data);
     MergePrefsDataWithPrepopulated();
