diff --git a/components/browser_ui/styles/android/java/src/org/chromium/components/browser_ui/styles/ChromeColors.java b/components/browser_ui/styles/android/java/src/org/chromium/components/browser_ui/styles/ChromeColors.java
index 127bf539cee81b9c32e740b0eb44f3d6db7ebe8e..56d75e7ed5daba1330b8f9f9a44340cd7d4a655e 100644
--- a/components/browser_ui/styles/android/java/src/org/chromium/components/browser_ui/styles/ChromeColors.java
+++ b/components/browser_ui/styles/android/java/src/org/chromium/components/browser_ui/styles/ChromeColors.java
@@ -29,9 +29,10 @@ public class ChromeColors {
      * @return The default theme color.
      */
     public static @ColorInt int getDefaultThemeColor(Context context, boolean isIncognito) {
-        return isIncognito
-                ? context.getColor(R.color.toolbar_background_primary_dark)
-                : MaterialColors.getColor(context, R.attr.colorSurface, TAG);
+        return context.getColor(R.color.toolbar_background_primary_dark);
+        // return isIncognito
+        //         ? context.getColor(R.color.toolbar_background_primary_dark)
+        //         : MaterialColors.getColor(context, R.attr.colorSurface, TAG);
     }
 
     /**
