diff --git a/components/crash/android/BUILD.gn b/components/crash/android/BUILD.gn
index a2d59905e4437d237c9651f040cd0c7a28b6ec17..555129e0a32294d0dfe56d04ebbd7e4241c36817 100644
--- a/components/crash/android/BUILD.gn
+++ b/components/crash/android/BUILD.gn
@@ -30,6 +30,7 @@ android_library("java") {
     "//content/public/android:content_main_dex_java",
     "//third_party/androidx:androidx_annotation_annotation_java",
     "//third_party/jni_zero:jni_zero_java",
+    "//net/android:net_java",
   ]
 
   srcjar_deps = [
