diff --git a/components/crash/android/java/src/org/chromium/components/crash/PureJavaExceptionReporter.java b/components/crash/android/java/src/org/chromium/components/crash/PureJavaExceptionReporter.java
index a2c2d22cc6a864a0b0ab5d2341ff3de3b5b62c3e..984c61988c95a51214adcf80e6af1db22c4b4d7a 100644
--- a/components/crash/android/java/src/org/chromium/components/crash/PureJavaExceptionReporter.java
+++ b/components/crash/android/java/src/org/chromium/components/crash/PureJavaExceptionReporter.java
@@ -25,6 +25,14 @@ import java.util.HashMap;
 import java.util.Map;
 import java.util.UUID;
 import java.util.concurrent.atomic.AtomicReferenceArray;
+import java.io.DataOutputStream;
+import java.net.HttpURLConnection;
+import java.io.FileInputStream;
+import java.net.URL;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import org.chromium.net.ChromiumNetworkAdapter;
+import org.chromium.net.NetworkTrafficAnnotationTag;
 
 /**
  * Creates a crash report and uploads it to crash server if there is a Java exception.
@@ -63,6 +71,9 @@ public abstract class PureJavaExceptionReporter
     private static final String RN = "\r\n";
     private static final String FORM_DATA_MESSAGE = "Content-Disposition: form-data; name=\"";
 
+    private static final String SENTRY_MINIDUMP_URL =
+        "https://o4509807793733632.ingest.de.sentry.io/api/4509807795372112/minidump/?sentry_key=591d5ee7a98b1fdeca311066ff238573";
+
     private boolean mUpload;
     protected Map<String, String> mReportContent;
     protected File mMinidumpFile;
