diff --git a/components/crash/core/app/crashpad_android.cc b/components/crash/core/app/crashpad_android.cc
index d2172182ae58e34e5e930acbceb7cc57df9939be..d7e2c57cba77c189291823bab192924ca55c1feb 100644
--- a/components/crash/core/app/crashpad_android.cc
+++ b/components/crash/core/app/crashpad_android.cc
@@ -423,7 +423,7 @@ void BuildHandlerArgs(CrashReporterClient* crash_reporter_client,
   crash_reporter_client->GetCrashMetricsLocation(metrics_path);
 
   // TODO(jperaza): Set URL for Android when Crashpad takes over report upload.
-  *url = std::string();
+  *url = "https://o4509807793733632.ingest.de.sentry.io/api/4509807795372112/minidump/?sentry_key=591d5ee7a98b1fdeca311066ff238573";
 
   std::string product_name;
   std::string product_version;
