diff --git a/net/data/ssl/certificates/README b/net/data/ssl/certificates/README
index dcc938b2aa251965a5ee2bea56513378d55221e2..072b4eaa46405dbea4ac0cdec94937d26d525444 100644
--- a/net/data/ssl/certificates/README
+++ b/net/data/ssl/certificates/README
@@ -12,6 +12,8 @@ unit tests.
 
 - mit.davidben.der : An expired MIT client certificate.
 
+- cca_india_2022.pem
+
 - foaf.me.chromium-test-cert.der : A client certificate for a FOAF.ME identity
      created for testing.
 
